### PR TITLE
Move physics simulations to fixed time step

### PR DESCRIPTION
**Pull request by Lunkers**
What's been done:
- Updates `Unity.Physics` package to version `0.5.1-preview.2`, which moves all simulations to a fixed time step, instead of updating every frame. This makes our physics framerate independent.
- Updates Systems that run alongside the physics simulation (`TargetScoringSystem` and `ResetThrowableSystem`) to also run on the fixed time step
- Updated the physics data. If you get issues related to physics data when trying to run the sample scene, try selecting Window > DOTS > Physics > Upgrade Data.
- Added some basic functionality to the sample scene. This is for debugging any physics system without needing to compile to a device.
- Added a small comment in `GameManager`. Which is where (I think) we need to write some code to ensure that the simulation update frequency matches the server tick rate for multiplayer.
- Made cups dynamic again. I haven't tested FPS with the fixed time step, so I'll leave them as static for now.

The PR is tested with the gameplay loop present on the current master branch

TL;DR: Moved physics to a fixed timestep. This is needed for multiplayer to work well, and makes the simulations deterministic.

### CODE REVIEW

**Comment by Lunkers** [comment]
New notes: Added graphics interpolation for the ball, hopefully this fixes stuttery motion on devices with high framerate (one of @bonnes1 , @jesperlundqvist , @matlar4 please verify, my phone is not fast enough).

Another note; seems like the updated burst version causes issues with android builds. I'll look into this today.
